### PR TITLE
show blockchain log entries on errors

### PR DESCRIPTION
**Pull request by euonymos**
When experimenting with failing tests I guess t would be beneficial to see the log. Alternatively we can add another helper which do the job, leaving `testNoErrors`.

### CODE REVIEW

**Comment by euonymos** [comment]
Looks like CI runs into OOM or something.

File: src/Plutus/Test/Model/Blockchain.hs [hunk 1]
```diff
@@ -390,6 +390,8 @@ data Blockchain = Blockchain
   , -- | human readable names. Idea is to substitute for them
     -- in pretty printers for error logs, user names, script names.
     bchNames :: !BchNames
+  , -- todo not the best place
+    stateDump :: Blockchain -> String
```
Comment (anton-k) [inline src/Plutus/Test/Model/Blockchain.hs:394]: No need for this function. It seems to be doing something that user can do

File: src/Plutus/Test/Model/Blockchain.hs [hunk 2]
```diff
@@ -843,20 +846,26 @@ waitNSlots n = modify' $ \s -> s {bchCurrentSlot = bchCurrentSlot s + n}
...
   updateCertificates
   updateFees
   saveTx
   saveDatums
+  saveState
   where
+
+    saveState = do
+      s <- get
+      logInfo $ stateDump s s
```
Comment (anton-k) [inline src/Plutus/Test/Model/Blockchain.hs:861]: User can do it if user needs that. No need to have forced log

File: src/Plutus/Test/Model/Fork/TxExtra.hs [hunk 3]
```diff
@@ -40,14 +41,15 @@ toExtra = Tx mempty
 data Extra = Extra
   { extra'withdraws      :: [Withdraw]
   , extra'certificates   :: [Certificate]
+  , extra'descr          :: Last String
```
Comment (anton-k) [inline src/Plutus/Test/Model/Fork/TxExtra.hs:44]: Let's keep `Tx` for things that relate to Cardano. This is test concept to improve readability. We can use `BchNames` for that or add this info to something that is not related to `Tx`. For example to Stats. But `BchNames` is preferable. It's the place to keep human readably synonyms for pretty printing.

**Review by anton-k — CHANGES_REQUESTED** [review]

**Comment by anton-k** [comment]
Thanks for your improvements to the lib so far. I will accept the PR if it pass CI. 
Fixing the lint job is ok. It seems that build fails by out of resources exception on CI-node

**Comment by anton-k** [comment]
Also thanks for docs update it's very helpful to keep it up to date.

**Comment by euonymos** [comment]
@anton-k I fixed lint tips, and reworked the prettifier, now it looks far better.
image

**Comment by anton-k** [comment]
Looks great!

**Comment by anton-k** [comment]
Maybe it would be cool to have option to show short versions of hashes, like github does for commits

**Pull request merged**